railties/test/generators/app_generator_test.rb

```diff
@@ -1042,6 +1042,35 @@ def test_version_control_initializes_git_repo
     assert_directory ".git"
   end
 
+  def test_default_branch_main_without_user_default
+    current_default_branch = `git config --global init.defaultBranch`
+    `git config --global --unset init.defaultBranch`
+
+    run_generator [destination_root]
+    assert_file ".git/HEAD", /main/
+
+    if !current_default_branch.strip.empty?
+      `git config --global init.defaultBranch #{current_default_branch}`
+    end
```

Comment:
this seems like something that should be inside an ensure block. otherwise a failing test would permanently remove a suite-runners's git config